docs/web_use.rst

```diff
@@ -1,2 +1,256 @@
 Web use
 =======
+
+Introduction
+............
+
+Let's see a simple way to use ``trafaret`` for handle endpoints of rest api in
+``aiohttp``.
+We'll create a simple rest api for create / update books. The first thing which
+we need to do it's describe our schemes of input data in trafaret format.
+
+When we work with form in the web we receive all data in a string format. So
+when you want to send a boolean type or list of integers you send some like this.
+
+.. code-block:: python
+
+    "True" # True 
+    "1, 2, 3" # [1, 2, 3]
+
+Trafaret designed for solve these problems.
+
+.. code-block:: python
+
+    import trafaret as t
+
+    def comma_to_list(text):
+        """Convert string with words separated by comma to list."""
+        return [
+            s.strip() for s in text.split(',')
+        ]
+
+    create_book_chacker = t.Dict({
+       'title': t.String,
+       'authors': comma_to_list,
+       'sold': t.StrBool,
+    })
+
+
+So, when you recive data from form it's not problem for you, because ``StrBool``
+and  ``comma_to_list`` prepare data for you in correct format.
+
+.. code-block:: python
+
+    create_book_chacker.check({"title": 'Glue', 'authors': 'Welsh,', 'sold': 'True'}) 
+    # {'title': 'Glue', 'authors': ['Welsh', ''], 'sold': True}
+
+But if you receive data as ``json`` it's not very useful for you, because you
+can receive data in correct format from the client.
+
+The second problem which trafaret solved it's a `camel/snake case war`.
+People why write in python prefe use `snake_case` unlike people why write in `ES`
+and use `CamelCase`. Trafaret give an approach for rename key of dictionary for
+solve this problem.
+
+.. code-block:: python
+
+    t.Dict({t.Key('userNameFirst') >> 'first_name': t.String}) 
+
+    # or
+
+    t.Dict({t.Key('userNameFirst', to_name='first_name'): t.String}) 
+
+Schemes
+.......
+
+So, now we are ready to write our schemes with `trafaret`. We can put this to
+the ``utils.py``.
+
+.. code-block:: python
+
+    import trafaret as t
+    from trafaret.keys import subdict
+
+
+    def author_validation(data):
+        """Test that first author exist in list of all authors."""
+        
+        if data['first_author'] not in data['authors']:
+            raise t.DataError('first_author must exist in authors')
+
```

Comment:
you should return something that will be stored in 'authors' key.

Reply:
But i don't want. I want only validate relation between two fields.

Reply:
Then you are using wrong instrument. subdict will remove both keys from result set and will return `'authors': None` instead.
To make just this check it is better to check it after, like `t.Dict(jsdkahfjkashfksajfh) & check_that_author_is_in_authors`.

Reply:
Okey, thx i'll try to rewrite it.